plugin-templates/features/class-featured-image-caption.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Featured_Image_Caption class file
+ *
+ * @package create-wordpress-plugin
+ */
+
+namespace Create_WordPress_Plugin;
+
+use Alley\WP\Types\Feature;
+
+final class Featured_Image_Caption implements Feature {
+	public function boot(): void {
+		add_filter( 'render_block_core/post-featured-image', [ $this, 'add_caption_to_featured_image' ], 10, 1 );
+	}
+
+	public function add_caption_to_featured_image( $block_content  ) {
+		if ( empty( get_the_ID() ) ) {
```

Comment:
Rather than calling `get_the_ID()`, I'd suggest that this callback receive the `WP_Block` parameter (the third one) and use the `postId` from the block context. Although they're likely to be the same ID in practice, using the passed context would reduce this feature's dependence on global state.